### ISSUE TITLE
On saving a file, I get: error org-roam--fix-relative-links: Buffer is read-only. Need to save twice to succeed

### ISSUE THREAD

**Issue by wilya7** [org-roam/org-roam#932]
### Description
The backup files are saved in a separate directory, to avoid spamming the files all over the place. This is in my init.el:
`setq backup-directory-alist '(("." . "~/.emacs.d/backups"))) `

With this setting, when saving an org-roam file I get the error listed below, if I save again it goes trough:
```
org-roam--fix-relative-links: Buffer is read-only: #<buffer !Users!cfusco!Nextcloud!work!zettelkasten!20200712212549-learning_improvement.org~>
Saving file /Users/cfusco/Nextcloud/work/zettelkasten/20200712212549-learning_improvement.org...
```
When I set backup-directory-alist to nil  (and the backup files stay in the org-roam directory) the error disappears, however in this setting  an other bug related to the backup files (ending with ~) becomes evident. If I link from document A to document B , than I go to B and I modify the text and I save, the back-link disappears because document A ends up pointing to the backup of document B (ending with ~), not to document B itself. To cope with the problem I disabled the creation of backup files with:
`(setq make-backup-files nil)`
This way the problem disappears.

Because both behaviors involve the backup files I believe it is the same bug, however if I am told otherwise I can open a separate bug report.

#### Steps to Reproduce
1. in init.el: 
`setq backup-directory-alist '(("." . "~/.emacs.d/backups")))`  
2. start emacs and load an org-roam file
3. modify the file and save
4. get the error

#### Backtrace

This is found in messages. I do not find any debugger buffer, even after giving  
`M-x toggle-debug-on error`

```
Saving file /Users/cfusco/Nextcloud/work/zettelkasten/20200712212549-learning_improvement.org...
org-roam--fix-relative-links: Buffer is read-only: #<buffer !Users!cfusco!Nextcloud!work!zettelkasten!20200712212549-learning_improvement.org~>
```
#### Expected Results

On saving a file there should be no error.

#### Actual Results

To save a file it is necessary to give the command twice.

### Environment
- Emacs: GNU Emacs 26.3 (build 1, x86_64-apple-darwin18.7.0, NS appkit-1671.60 Version 10.14.6 (Build 18G5033))
 of 2020-06-16

**Comment by jethrokuan** [comment]
This sounds like it would've been resolved by #915 , are you on a commit that has this fix?

**Comment by wilya7** [comment]
Duplicate of #915 

@jethrokuan, thank you so much. This bug was driving me crazy. Indeed it #915 issue. Thanks again.
